docs/config-luauconfig.md

```diff
@@ -0,0 +1,144 @@
+    | "UnknownGlobal"
+    | "DeprecatedGlobal"
+    | "GlobalUsedAsLocal"
+    | "LocalShadow"
+    | "SameLineStatement"
+    | "MultiLineStatement"
+    | "LocalUnused"
+    | "FunctionUnused"
+    | "ImportUnused"
+    | "BuiltinGlobalWrite"
+    | "PlaceholderRead"
+    | "UnreachableCode"
+    | "UnknownType"
+    | "ForRange"
+    | "UnbalancedAssignment"
+    | "ImplicitReturn"
+    | "DuplicateLocal"
+    | "FormatString"
+    | "TableLiteral"
+    | "UninitializedLocal"
+    | "DuplicateFunction"
+    | "DeprecatedApi"
+    | "TableOperations"
+    | "DuplicateCondition"
+    | "MisleadingAndOr"
+    | "CommentDirective"
+    | "IntegerParsing"
+    | "ComparisonPrecedence"
+    | "RedundantNativeAttribute"
+
+type LuauConfig = {
+    languagemode: LanguageMode?,
+    lint: { [LintWarning]: boolean }?,
+    linterrors: boolean?,
+    typeerrors: boolean?,
+    globals: { string }?,
+    aliases: { [string]: string }?,
+}
+
+type Config = {
+    luau: LuauConfig?
+}
+```
+
+If configuration fields are not provided, `.config.luau` has the same default behavior as `.luaurc`.
+
+### Search semantics
+
+The search semantics of `.config.luau` files are the same as `.luaurc`'s (adapted from a previous RFC):
+
+> For a given `.luau` or `.lua` script, Luau will search for `.luaurc` files starting from the folder that the script is in; all files in the ancestry chain will be parsed and their configuration applied.
+> When multiple configuration files exist throughout the ancestry chain, configurations closer to the script override those in higher-level directories.
+
+If both `.luaurc` and `.config.luau` are present in a directory, an error is thrown.
+
+### Configuration extraction
+
+Unlike `.luaurc` files, `.config.luau` files can contain runtime constructs like variables, functions, and loops.
+To extract configuration from a `.config.luau` file, its contents are executed in an isolated Luau VM with only the standard libraries enabled (`luaL_openlibs`).
+This approach ensures that configuration files can leverage Luau features while maintaining a sandboxed execution environment.
+
+#### Timing out
+
+Executing arbitrarily complex Luau code requires us to guard against excessive computation:
+```luau
+-- .config.luau with an infinite loop
+return (function()
+    while true do
+    end
+end)()
+```
+
+For analysis-time configuration extraction, we adopt the same behavior described in the [user-defined type functions RFC](https://rfcs.luau.org/user-defined-type-functions.html): we simply respect the time limit that the embedding context provides through the `Frontend` API.
+
+At runtime, the configuration extraction timeout for each `.config.luau` file is set to two seconds by default.
+This duration is generally sufficient for typical configuration logic while being short enough to prevent accidental or malicious long-running scripts from impacting performance.
+For more fine-grained control, however, embedders will be free to override this.
```

Comment:
```suggestion
For more fine-grained control, however, embedders will be free to override this. That is to say, we do not wish to prescribe that this limit in general, but note that a limit being present in general is important.
```

Maybe a bit too legalistic, but I sort of want to explicitly say that we don't consider it wrong to have a different limit, and we're really just saying a limit must be present at runtime.